videostore/utils.go

```diff
@@ -229,41 +263,85 @@ func formatDateTimeToString(dateTime time.Time) string {
 
 // matchStorageToRange returns a list of files that fall within the provided time range.
 // Includes trimming video files to the time range if they overlap.
-func matchStorageToRange(files []string, start, end time.Time, logger logging.Logger) []string {
+// The matcher assumes that the input files list is sorted by start time and the underylying
+// video segments do not overlap.
+func matchStorageToRange(files []fileWithDate, start, end time.Time, logger logging.Logger) []string {
```

Comment:
can we change this to return a struct that represents the .txt file & then have that struct have a String() method that can print the file.

The reason being that it will remove the need to format strings in this code, and it will make it easier to write tests for this function without needing to worry about conforming to the txt file format.

Reply:
Created struct `concatFileEntry`